src/main/java/com/github/manolo8/darkbot/config/ConfigManager.java

```diff
@@ -103,22 +107,34 @@ private Config loadConfig(Path configFile, Path backupFile) {
     public IDarkBotAPI getAPI(StartupParams params) {
         if (params.useNoOp()) return new NoopApiAdapter();
         try {
+
             if (config.BOT_SETTINGS.API_CONFIG.API == 0) return new DarkBotApiAdapter();
             else if (config.BOT_SETTINGS.API_CONFIG.API == 1) return new DarkFlashApiAdapter(params);
             else if (config.BOT_SETTINGS.API_CONFIG.API == 2) return new DarkBoatAdapter(params);
             else if (config.BOT_SETTINGS.API_CONFIG.API == 3) return new NativeApiAdapter(params);
             else if (config.BOT_SETTINGS.API_CONFIG.API == 4) return new NoopApiAdapter();
             else throw new IllegalArgumentException("API not found: " + config.BOT_SETTINGS.API_CONFIG.API);
         } catch (Error e) {
-            System.out.println("Error enabling API #" + config.BOT_SETTINGS.API_CONFIG.API + ", using no-op api");
             e.printStackTrace();
-            config.BOT_SETTINGS.API_CONFIG.API = 4;
-            Popups.showMessageAsync(
-                    "API failed to load",
-                    "The API you had selected is not able to load.\n" +
-                    "You probably do not have the required DLL in your lib folder.\n" +
-                    "The bot will start on no-operation API, change it in the settings and restart.",
-                    JOptionPane.ERROR_MESSAGE);
+
+            APIErrors err = new APIErrors();
+
+            final String WORKING_DIR  = System.getProperty("user.dir");
+            final String ERR_MSG      = e.getMessage();
+            final String JAVA_32_ERR  = "not found in resource path";
+
+            if      (ERR_MSG.contains(JAVA_32_ERR))          err.displayUnsupportedJavaError();
+            else if (!WORKING_DIR.matches("[\\x00-\\x7F]+")) err.displayInvalidPathError(WORKING_DIR);
```

Comment:
Lastly, a suggested above, make the actual checking/deciding what error you show hidden away inside ApiErrors, none of that logic should be here